src/ert/_c_wrappers/enkf/ert_config.py

```diff
@@ -547,33 +596,52 @@ def _workflows_from_dict(cls, content_dict, substitution_list):
                 )
 
         for hook_name, mode_name in hook_workflow_info:
-            if mode_name not in [runtime.name for runtime in HookRuntime.enums()]:
-                raise ConfigValidationError(
-                    errors=[f"Run mode {mode_name!r} not supported for Hook Workflow"]
+            try:
+                run_mode = HookRuntime.from_string(mode_name)
+            except ValueError:
+                combined_error.add_error(
+                    ConfigValidationError(
+                        errors=f"Run mode {mode_name!r} not supported for Hook "
+                        f"Workflow"
+                    )
                 )
+                continue
 
-            if hook_name not in workflows:
-                raise ConfigValidationError(
-                    errors=[
-                        f"Cannot setup hook for non-existing job name {hook_name!r}"
-                    ]
+            try:
+                workflow = workflows[hook_name]
+            except KeyError:
+                combined_error.add_error(
+                    ConfigValidationError(
+                        errors=f"Cannot setup hook for non-existing jo"
```

Comment:
This error message was actually always wrong, could we change it now and just have a separate commit for it? IT should say `non-existing workflow` instead of `non-existing job`.

Reply:
Done